docs/paper/dev/api/how-do-plugins-work.md

```diff
@@ -0,0 +1,100 @@
+---
+slug: /dev/how-do-plugins-work
+description: How plugins work in Paper.
+---
+
+# How Plugins Work
+
+## Introduction
+
+Plugins are a way to extend the functionality of a Minecraft server. They are written in Java and can be loaded and 
+unloaded at runtime. Plugins are loaded from the `plugins` folder in the server directory. Plugins can be loaded from a 
+`.jar` file or from a folder. 
+
+## Plugin Lifecycle
+
+Plugins are loaded and unloaded at runtime. When a plugin is loaded, it is initialized and enabled. When a plugin is 
+unloaded, it is disabled and finalized.
+
+### Initialization
+
+When a plugin is loaded, it is initialized. This means that the plugin is loaded into memory and its `onLoad` 
+method is called. This method is used to initialize the plugin and set up any resources that it needs.
+
+### Enabling
+
+When a plugin is enabled, its `onEnable` method is called. This method is used to set up any resources that the plugin 
+needs to run. This method is called when the plugin is initialized but before the server has started ticking, so it is 
+safe to register event listeners and other resources that the plugin needs to run, however often not safe to interact 
+with a lot of APIs.
+
+### Disabling
+
+When a plugin is disabled, its `onDisable` method is called. This method is used to clean up any resources that the 
+plugin has allocated. This method is called before all plugins are unloaded, and is meant for any cleanup that needs to
+be done before the plugin is unloaded.
```

Comment:
worthwhile to mention some examples as with `onEnable`, such as closing database connections or saving data